Reorganize account deletion docs (#1843)
gruntwork-io/docs: _docs-sources/foundations/landing-zone/manage-accounts.md

### Remove an AWS account

We recommend the following three step procedure to close an AWS account managed by Devops Foundations:

1. [Cleanup the OpenTofu/Terraform State for the Control Tower module](#1-cleanup-opentofuterraform-state-for-the-control-tower-module)
1. [Cleanup Infrastructure Code](#2-cleanup-infrastructure-code)
1. [Close the Account with Clickops](#3-close-the-accounts-in-aws-organizations)

#### 1. Cleanup the OpenTofu/Terraform State for the Control Tower module

1. Obtain AWS CLI credentials for the your *management* account.
1. Remove the targeted account(s) from being controlled by the *management* account's Control Tower module by running the following commands:

  ```bash
  cd management/_global/control-tower-multi-account-factory/ # Navigate to the Control Tower module directory in your central/root infrastructure repository

>   terragrunt plan # Observe that targeted accounts are set for destruction

Reviewer: @oredavids if this is step 1, then the plan would show no changes, right?  Do we need to plan/apply at all here?

Author: I realized that splitting the state modification from Iac cleanup is not ideal, this has been shuffled back into IaC cleanup and the comments make sense again.

### Remove an AWS account

We recommend the following three step procedure to close an AWS account managed by Devops Foundations:

1. [Cleanup the OpenTofu/Terraform State for the Control Tower module](#1-cleanup-opentofuterraform-state-for-the-control-tower-module)
1. [Cleanup Infrastructure Code](#2-cleanup-infrastructure-code)
1. [Close the Account with Clickops](#3-close-the-accounts-in-aws-organizations)

> Note that we are recommending that ultimately accounts be closed with ClickOps instead of using Gruntwork Pipelines. Removing the account via pipelines by deleting the account request file can and often does work, however the underlying AWS Service Catalog that we use to interact with Control Tower and deprovision the account is not reliable and often returns spurious errors that can require multiple retries to complete successfully.  The procedure here is fundamentally about working around that unreliability.

Reviewer: ```suggestion
We are recommending that accounts be closed with ClickOps instead of using Gruntwork Pipelines. Removing the account via pipelines by deleting the account request file can and often does work, however the underlying AWS Service Catalog that we use to interact with Control Tower and deprovision the account is [not reliable](https://github.com/hashicorp/terraform-provider-aws/issues/31705) and often returns spurious errors that can require multiple retries to complete successfully.  The procedure here is fundamentally about working around that unreliability.
```

#### 1. Cleanup the OpenTofu/Terraform State for the Control Tower module

1. Obtain AWS CLI credentials for the your *management* account.
1. Remove the targeted account(s) from being controlled by the *management* account's Control Tower module by running the following commands:

  ```bash
  cd management/_global/control-tower-multi-account-factory/ # Navigate to the Control Tower module directory in your central/root infrastructure repository
  terragrunt plan # Observe that targeted accounts are set for destruction
  terragrunt state rm 'module.accounts["<ACCOUNT_NAME>"].aws_servicecatalog_provisioned_product.control_tower_factory' # Optionally, use 'rm -dry-run' to preview the removal.
  terragrunt plan # Verify no accounts are set for destruction but outputs are updated
  terragrunt apply # Apply the changes.
  ```

#### 2. Cleanup Infrastructure Code

> All code changes should be committed for each repository, **at the end**, to Git with a **[skip ci]** in the commit message to avoid triggering Gruntwork Pipelines CI processes.

Reviewer: ```suggestion
All the code changes recommended here should be committed for each repository, **at the end**, to Git with a **[skip ci]** in the commit message to avoid triggering Gruntwork Pipelines CI processes.
```

    This is a plan/apply operation because the TGW module reads the `accounts.yml` file to determine which accounts are attached to the TGW.

1. Delete the folders for the targeted accounts from the repository.
1. Commit all changes to your branch, ensuring to include `[skip ci]` in the commit message.
1. Open a Pull Request and observe that `Pipelines Plan` is absent.
1. Approve and **squash-merge**(if you have multiple commits) the Pull Request, again ensuring to include `[skip ci]` in the commit message to prevent Gruntwork Pipelines from initiating any destruction processes.

##### Delegated Infrastructure Repository (Enterprise-only)

> If targeted accounts were created as part of a separate infra-live-team repository and;

Reviewer: ```suggestion
If targeted accounts were created as part of a separate delegated infra-live repository and;
```

> 1. If all the accounts in a team-repo have been closed; delete the entire repository else delete the accounts folders for only the closed accounts and repeat the CI steps of creating a PR and merging it with `[skip ci]` in the commit message.

Reviewer: ```suggestion
1. If all the accounts in a delgated repo have been closed; delete the entire repository else delete the accounts folders for only the closed accounts and repeat the CI steps of creating a PR and merging it with `[skip ci]` in the commit message.
```

> 2. If team repositories were removed above **and** you have a setup that includes an `infrastructure-pipelines` repository; complete the cleanup process by removing the deleted repository references from the Gruntwork Pipelines configuration in `.gruntwork/config` file located in the `infrastructure-pipelines` repository via a Pull Request and merge it into the designated *main* branch.

Reviewer: ```suggestion
2. If delegated repositories were removed above **and** you have a setup that includes an `infrastructure-pipelines` repository; complete the cleanup process by removing the deleted repository references from the Gruntwork Pipelines configuration in `.gruntwork/config` file located in the `infrastructure-pipelines` repository via a Pull Request and merge it into the designated *main* branch.
```
